source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1357,8 +1334,8 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
           absl::FailedPreconditionError("unhandled immediate response due to config disabled it");
 
       if (on_processing_response_) {
-        on_processing_response_->afterReceivingImmediateResponse(*response, processing_status,
-                                                                 decoder_callbacks_->streamInfo());
+        on_processing_response_->afterReceivingImmediateResponse(
```

Comment:
In this case, the immediate response handling is disabled by filter configuration, i.e, the immediate response will be taken as server malfunction, and the response will be rejected.  Is there ay reason to store such kind of immediate response?

Reply:
The config allows for saving of the response even if the `processing_status` is an [error](https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/http/ext_proc/response_processors/save_processing_response/v3/save_processing_response.proto#L34). However, in this case, it probably makes sense to not save the immediate response. Updated.